library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -423,6 +433,9 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
     if (captionDataProcessed) {
       if (captionMode == CC_MODE_ROLL_UP || captionMode == CC_MODE_PAINT_ON) {
         cues = getDisplayCues();
+        if (!captionEraseCommandSeen) {
+          ccTimeOutCounter = System.currentTimeMillis();
```

Comment:
Ah: And I think we should use `SystemClock.elapsedRealtime()` rather than `System.currentTimeMillis()` just so we don't need to worry about the clock changing underneath us.

More info on the difference:
https://developer.android.com/reference/android/os/SystemClock

Reply:
Yes, you mean use `Clock.elapsedRealtime()` right?  The testable [Clock](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/util/Clock.html) interface?  Which I agree, is a great idea.

Reply:
Sorry - yeah.

We should: a) use Clock and b) use `elapsedRealtime()`

Reply:
@icbaker Thank you for the feed back. Modified 
- Cea608Decoder to use com.google.android.exoplayer.util.Clock.
- The timeout is. now passed to  Cea608Decoder constructor
- The ccTimeOutCounter is changes lastCueUpdateMs

Reply:
@icbaker Please let us know if there is any other changes needed